src/zeit/push/interfaces.py

```diff
@@ -244,6 +247,34 @@ class MobileButtonsSource(zeit.cms.content.sources.XMLSource):
 MOBILE_BUTTONS_SOURCE = MobileButtonsSource()
 
 
+class PayloadTemplateSource(zc.sourcefactory.basic.BasicSourceFactory):
+
+    # For testing purposes, this should not be cached at all
+    # @CONFIG_CACHE.cache_on_arguments()
+    def getValues(self):
+        payload_path = zope.app.appsetup.product.getProductConfiguration(
+            'zeit.push').get('push-payload-templates')
+        template_folder = zeit.cms.interfaces.ICMSContent(
+            "http://xml.zeit.de/{}".format(payload_path))
+        return [template_json for template_json in template_folder.values()]
+
+    def getTitle(self, value):
+        return value.__name__.split('.')[0].capitalize()
+
+    def getToken(self, value):
+        return value.uniqueId
+
+    def find(self, id):
```

Comment:
Also wenn wir die uniqueId in der message_config speichern, dann implementiert sich das hier flotter als `return ICMSContent(id, None)`. :) Aber wir sollten glaube ich lieber nur den Dateinamen speichern, weil dann der Pfad nicht in jedem Artikel drinsteht. Dann implementiert sich das als `folder.get(id, None)` (oder `ICMSContent(folder_id + id, None)`, wenn wir in der Konfig nen trailing slash verlangen :).